framework-plugins/lisk-framework-report-misbehavior-plugin/src/db.ts

```diff
@@ -83,3 +102,33 @@ export const saveBlockHeaders = async (
 	}
 	return false;
 };
+
+export const getContradictingBlockHeader = async (
+	db: KVStore,
+	blockHeader: BlockHeader,
+	schemas: RegisteredSchema,
+): Promise<BlockHeader | undefined> =>
+	new Promise((resolve, reject) => {
+		const stream = db.createReadStream({
+			gte: getFirstPrefix(blockHeader.generatorPublicKey.toString('binary')),
+			lte: getLastPrefix(blockHeader.generatorPublicKey.toString('binary')),
+		});
+		let conflictingHeader: BlockHeader | undefined;
+		stream
+			.on('data', ({ value }: { value: Buffer }) => {
+				const { blockHeaders } = codec.decode<BlockHeaders>(blockHeadersSchema, value);
+				for (const encodedHeader of blockHeaders) {
+					const decodedBlockHeader = decodeBlockHeader(encodedHeader, schemas);
+					// If there are multiple contradicting blocks, it will report the one with highest height
+					if (areHeadersContradicting(blockHeader, decodedBlockHeader)) {
+						conflictingHeader = decodedBlockHeader;
```

Comment:
Should not we close the stream here? Why to parse the tons of blocks if we already found on conflicting?

Reply:
yes, i wanted to close the stream here, but there are no interface to close in the middle

Reply:
The combination of these two may help. 

https://nodejs.org/api/stream.html#stream_readable_pause
https://nodejs.org/api/stream.html#stream_readable_destroy_error

So you pause the stream that will immediately stop emitting new `data` events and then you can destroy that will eventually release all resources and close.